### PR TITLE
Scripts/Commands: command .modify power (all powers)

### PR DESCRIPTION
**Pull request by Hadifaren**
`.modify power` `#powercount`

Closes #19288

**Target branch(es):** 3.3.5/master

- [x] master

**Tests performed:** (Does it build, tested in-game, etc.)

Tested And Works

### CODE REVIEW

**Comment by tkrokli** [comment]
```
- "sql/updates/auth/2017_11_16_20833_auth.sql" is missing in the 'auth'.'updates' table.
error:
  The Database Updater is broken for the 'auth' database,
  because the applied update is missing in the 'auth'.'updates' table.
How to fix:
  Insert the missing names of the already applied sql updates
  to the 'updates' table of the 'auth' base dump ('sql/base/auth_database.sql').
```

**Comment by Hadifaren** [comment]
i dont know how to solve the error you writed ^_^
+ cannot understand how to fix

**Comment by Hadifaren** [comment]
:| why its closed ! @tkrokli

File: src/server/scripts/Commands/cs_modify.cpp [hunk 1]
```diff
@@ -997,6 +998,66 @@ class modify_commandscript : public CommandScript
...
     }
+    	// Edit Player Power 
+	static bool HandleModifyPowerCommand(ChatHandler* handler, const char* args)
+	{
+		int32 power, maxpower;
+		Player* target = handler->getSelectedPlayerOrSelf();
+		int8 const powerMultiplier = 10;
+		if (CheckModifyResources(handler, args, target, power, maxpower, powerMultiplier))
+		{
+			switch (target->getPowerType())
```
Comment (tkrokli) [inline src/server/scripts/Commands/cs_modify.cpp]:
```
src/server/scripts/Commands/cs_modify.cpp:1009:12:
 fatal error:
 13 enumeration values not handled in switch: 'POWER_COMBO_POINTS', 'POWER_RUNES', 'POWER_SOUL_SHARDS'... [-Wswitch]
                        switch (target->getPowerType())
                                ^
1 error generated.
```
https://github.com/TrinityCore/TrinityCore/blob/2d75d8071bb6d85987bdc5d88ed2271822fcbcfd/src/server/game/Miscellaneous/SharedDefines.h#L250-L274

There are 22 `Powers` listed in the enum in `SharedDefines.h` which means you need to find a way to tell this script to skip/ignore the 13 powers not handled by this switch. I don't know what that code should look like, but maybe some of the other developers will know.

Comment (Jildor) [reply src/server/scripts/Commands/cs_modify.cpp]: Need add a default in switch and add tabs in  cases

Comment (tkrokli) [reply src/server/scripts/Commands/cs_modify.cpp]: OK, I get what you mean that it needs a
```cpp
			default:
				break;
```
at the end, but not sure what you mean by `add tabs in cases`

**Comment by Jildor** [comment]
Add tabulations in  all cases:
```diff
-    case X
+        case X
```

Sorry mobile :smile:

File: src/server/scripts/Commands/cs_modify.cpp (hunk 1 above)
Comment (Jildor) [reply src/server/scripts/Commands/cs_modify.cpp]: Add tabulations in  all cases:
```diff
-    case X
+        case X
```

Sorry mobile :smile:

Comment (tkrokli) [reply src/server/scripts/Commands/cs_modify.cpp]: OK, I see. Add correct indent to each case,

Comment (tkrokli) [reply src/server/scripts/Commands/cs_modify.cpp]: BTW; changes have been made, but they are all below this comment section
(and hence they don't get hidden by the update).

**Comment by tkrokli** [comment]
OK - so far, so good. The C++ part compiles without errors.
Only the `sql/base/auth_database.sql` part left to fix to have travis-ci show up green.

**Comment by tkrokli** [comment]
@Hadifaren : I hope you take notes on my changes so you can use that knowledge later. 😃

**Comment by Hadifaren** [comment]
@tkrokli of course your comments are so helpful ^_^

**Comment by Hadifaren** [comment]
of course your comments always are helpful ^_^

**Comment by Seyden** [comment]
A player can have multiple powers.. what you do then? What if i want to change the power that i got after i used a spell? With your change, it only uses my current power.

Also the power multiplier isn't always the same for each power type.

**Comment by tkrokli** [comment]
@Seyden : Looks like your question is answered in the closing commit (reading the added content & commit message). 😄

**Pull request closed without merging**